nmt.go

```diff
@@ -193,20 +265,35 @@ func (n *NamespacedMerkleTree) buildRangeProof(proofStart, proofEnd int) [][]byt
 		// reached a leaf
 		if end-start == 1 {
 			leafHash := n.leafHashes[start]
-			// if current range does not overlap with proof range, add a node to proofs
+			// if the index of the leaf node is out of the queried range i.e.
+			// , [proofStart, proofEnd)
+			// and if the leaf is required as part of the proof i.e., includeNode == true
 			if (start < proofStart || start >= proofEnd) && includeNode {
+				// add the leafHash to the proof
 				proof = append(proof, leafHash)
 			}
+			// if the index of the leaf is within the queried range i.e., [proofStart, proofEnd] OR
+			// if the leaf is not required as part of the proof i.e., includeNode == false
 			return leafHash
 		}
 
-		// recursively get left and right subtree
+		// newIncludeNode indicates whether one of the subtrees of the current subtree [start, end)
+		// may have an overlap with the queried proof range i.e., [proofStart, proofEnd)
 		newIncludeNode := includeNode
+		// check whether the subtree representing the [start, end) range of leaves has overlap with the
+		// queried proof range i.e., [proofStart, proofEnd)
+		// if not
```

Comment:
was the rest of this comment lost?

Reply:
No, it was not lost, `if not` refers to the following `if` statement which is entered when the said condition is not met. I tried to clarify it in this commit https://github.com/celestiaorg/nmt/commit/26b1ea676ac264bb2c6894efcd7a4f4aaf3924b3.